File: models/MessageModel.js
const mongoose = require('mongoose');

const messageSchema = new mongoose.Schema({
    _id: {
        type: mongoose.Schema.Types.ObjectId,
        auto: true
      },
    idSender: { type: String, required: false },
    idRecipient: { type: String, required: false },
    idAdministrateur: { type: String, required: false },
    coverPicture:{type:String,required:false,
        default:"https://img.freepik.com/vecteurs-libre/caractere-couple-tenant-nouvelle-illustration-message_53876-26822.jpg?w=740&t=st=1694366817~exp=1694367417~hmac=b6b118a22c685adad59d1f18d81747fe2e21062cc21c48c6a0520f4b3250d39e"
    },
    sender: {
        type: String,
        required: false,
    },
    recipient: {
        type: String,
        required: false,
    },
    subject: {
        type: String,
        required: false,
    },
    content: {
        type: String,
        required: false,
    },
    timestamp: {
        type: Date,
        default: Date.now,
    },
    read: {
        type: Boolean,
        default: false,
    },
}, {timestamps:true});


const MessageModel = mongoose.model('Message', messageSchema);

module.exports = MessageModel;

File: lib/src/model/MessageModel.dart
import  "package:flutter/material.dart";


class MessageModel {
  String? id;
  String? idSender;
  String? idRecipient;
  String? idAdministrateur;
  String? coverPicture;
  String? sender;
  String? recipient;
  String? subject;
  String? content;
  DateTime? timestamp;
  bool? read;
  DateTime? createdAt;
  DateTime? updatedAt;



  MessageModel({
    this.id,
    this.idSender,
    this.idRecipient,
    this.idAdministrateur,
    this.coverPicture,
    this.sender,
    this.recipient,
    this.subject,
    this.content,
    this.timestamp,
    this.read,
    this.createdAt,
    this.updatedAt,
  });

  factory MessageModel.fromJson(Map<String, dynamic> json) {
    return MessageModel(
      id: json['_id'],
      idSender: json['idSender'],
      idRecipient: json['idRecipient'],
      idAdministrateur: json['idAdministrateur'],
      coverPicture: json['coverPicture'],
      sender: json['sender'],
      recipient: json['recipient'],
      subject: json['subject'],
      content: json['content'],
      timestamp: DateTime.parse(json['timestamp']),
      read: json['read'],
      createdAt: DateTime.parse(json['createdAt']),
      updatedAt: DateTime.parse(json['updatedAt']),
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'idSender': idSender,
      'idRecipient': idRecipient,
      'idAdministrateur': idAdministrateur,
      'coverPicture': coverPicture,
      'sender': sender,
      'recipient': recipient,
      'subject': subject,
      'content': content,
      'timestamp': timestamp?.toIso8601String(),
      'read': read,
      'createdAt': createdAt?.toIso8601String(),
      'updatedAt': updatedAt?.toIso8601String(),
    };
  }
}
